snapcraft.yaml

```diff
@@ -0,0 +1,37 @@
+name: dvc
+summary: Data Version Control
+description: Git for Data & Models https://dvc.org
+adopt-info: dvc
+grade: stable
+confinement: classic  # for same reasons as snap `git-ubuntu`
+base: core18
+license: Apache-2.0
+parts:
+  dvc:
+    plugin: python
+    python-packages: [wheel]
+    source: .
+    build-packages:
+    - git
+    - libc-dev
+    - libkrb5-dev  # gssapi build
+    stage-packages:
+    - git
+    override-pull: |
+        snapcraftctl pull
+        echo 'PKG = "snap"' > $SNAPCRAFT_PART_SRC/dvc/utils/build.py
+        snapcraftctl set-version `cd $SNAPCRAFT_PART_SRC && git describe --tags`
+        # install all optional extras
+        sed -ri 's/(=install_requires)/\1+all_remotes+ssh_gssapi+hdfs/' $SNAPCRAFT_PART_SRC/setup.py
+    override-build: |
+        snapcraftctl build
+        cp $SNAPCRAFT_PART_BUILD/scripts/completion/dvc.bash $SNAPCRAFT_PART_INSTALL/completion.sh
```

Comment:
`completion.sh` seems to propagate through the rest of the lifecycle (`stage` and `prime`) but this may stop working in future depending on what the `plugin: python` does. It's not documented anywhere AFAIK.